src/library/scala/runtime/LazyRef.scala

```diff
@@ -1,52 +1,157 @@
+/*                     __                                                      *\
+**     ________ ___   / /  ___     Scala API                                   **
+**    / __/ __// _ | / /  / _ |    (c) 2002-2016, LAMP/EPFL and Lightbend, Inc **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/                      **
+** /____/\___/_/ |_/____/_/ | |                                                **
+**                          |/                                                 **
+\*                                                                             */
+
 package scala.runtime
 
-/* Classes used as holders for local lazy vals */
+/** Classes used as holders for lazy vals defined in methods. */
 
 class LazyRef[T] {
-  var value: T = _
-  @volatile var initialized: Boolean = false
+  @volatile private[this] var _initialized: Boolean = _
```

Comment:
True, the runtime package is not meant for public consumption. Good point
that they should likely be final. They have to be public though, as they
are referenced from compiler-generated code.

https://github.com/scala/scala-dev/issues/339